### ISSUE TITLE
Creating an environment from the dev portal doesn't use the manifest default values

### ISSUE THREAD

**Issue by SimonDraime** [Azure/deployment-environments#34]
Hi Team,

As the parameters' default values of the ARM template I use in catalog are defined for a production environment, I set different default values in the manifest.yaml file.
However, when I create an environment from the dev portal it is using the default values of the ARM template instead of the ones defined in the manifest.

It is quite counterintuitive as it is not the values displayed in the dev portal during creation that are actually used.

Is it the expected behaviour? Is it possible to improve that behaviour as it would be very useful to be able to use different manifest files with the same ARM template for different environment type?

**Comment by j-rewerts** [comment]
Hey @SimonDraime. Thanks for the report! This is definitely a bug, we should be honoring what is defined as the default in the YAML file. I'll let you know when we've got a fix ready.

**Comment by j-rewerts** [comment]
Though I am curious to understand more about your last part:
`it would be very useful to be able to use different manifest files with the same ARM template for different environment type?`

We don't currently have this ability in the product. Mind expanding a bit more on what you're hoping to achieve?

**Comment by SimonDraime** [comment]
Yes, it would be a good idea to use the environment type to define some parameters that are always the same for a specific environment type.
However, where would you define these parameters' values on the Azure portal, it's not very clear for me how are linked the catalogs and the environment definitions to the environment types? What if you have several environment definitions in your catalog? How to defines the parameters' value for each environment definition in the environment type?

**Comment by SimonDraime** [comment]
@j-rewerts Do you have an idea of the timeframe for the fix availability?

**Comment by j-rewerts** [comment]
Nothing to report currently, I'll follow up once we're getting close to shipping the fix.

**Comment by ericaguthan** [comment]
Hey @timwebster9,

Thanks for the feedback! To clarify here, we have resolved the issue where defaults in the environment.yaml were not being passed through, which is the initial problem presented in this issue. The problem you describe (different defaults in the yaml and the IAC template) is one that we recommend solving by prioritizing setting defaults at the environment.yaml level. The yamls are what inform what the devportal users have the power to change. 

Hope this helps!

**Comment by ericaguthan** [comment]
Closing since this issue was not updated after the main issue was resolved. Confirmed "default parameters per environment type" is a recorded feature request in our backlog.

**Comment by ericaguthan** [comment]
Thank you for bumping this bug. We are taking a look at it and will update you once a solution is available.

**Comment by timwebster9** [comment]
Hi @ericaguthan 

Sorry I'm not clear. I am referring to the defaults in the environment.yaml.  If defaults are present and the user just accepts them without changing them, they still don't get passed through to the IaC.

Maybe this is just a problem when using Terraform?

To reproduce:

- Create a Terraform environment
- include variables in your `variables.tf` file with no default value
- include a corresponding parameter for the variable in your `environment.yaml` file.  Give the parameter a default value.
- Try to create an environment without changing the default value.  It should fail with a Terraform error saying the variable is empty and has no default value.

I'm happy to open a separate issue if that would help?

**Comment by ericaguthan** [comment]
Hi @timwebster9,

This issue is fine to leave open. You are correct that this bug exists and is not just for Terraform. Expected behavior is that the environment yaml is the source of truth so what users see in the DevPortal is what they get. We are taking a look and will update when the fix has hit all regions.
